client/my-sites/site-settings/spam-filtering-settings.jsx

```diff
@@ -17,40 +17,45 @@ import FormFieldset from 'components/forms/form-fieldset';
 import FormLabel from 'components/forms/form-label';
 import FormTextInput from 'components/forms/form-text-input';
 import FormInputValidation from 'components/forms/form-input-validation';
+import FormSettingExplanation from 'components/forms/form-setting-explanation';
 import Gridicon from 'components/gridicon';
 import SupportInfo from 'components/support-info';
 import ExternalLink from 'components/external-link';
-import { getSelectedSiteId } from 'state/ui/selectors';
+import { isJetpackAntiSpam } from 'lib/products-values';
+import { getSelectedSiteId, getSelectedSiteSlug } from 'state/ui/selectors';
+import { isFetchingSitePurchases } from 'state/purchases/selectors';
 import isJetpackSettingsSaveFailure from 'state/selectors/is-jetpack-settings-save-failure';
-import FormSettingExplanation from 'components/forms/form-setting-explanation';
+import { getSiteProducts } from 'state/sites/selectors';
 import { hasFeature } from 'state/sites/plans/selectors';
-import { shouldShowOfferResetFlow } from 'lib/plans/config';
 import {
 	FEATURE_SPAM_AKISMET_PLUS,
 	FEATURE_JETPACK_ANTI_SPAM,
 	FEATURE_JETPACK_ANTI_SPAM_MONTHLY,
-	PLAN_JETPACK_PERSONAL,
 } from 'lib/plans/constants';
+import { PRODUCT_JETPACK_ANTI_SPAM } from 'lib/products-values/constants';
 
 const SpamFilteringSettings = ( {
 	currentAkismetKey,
 	dirtyFields,
 	fields,
 	hasAkismetFeature,
 	hasAkismetKeyError,
-	hasAntiSpam,
+	hasAntiSpamFeature,
+	hasJetpackAntiSpamProduct,
 	isRequestingSettings,
+	isRequestingSitePurchases,
 	isSavingSettings,
 	onChangeField,
+	siteSlug,
 	translate,
 } ) => {
 	const { akismet: akismetActive, wordpress_api_key } = fields;
-	const isStoredKey = wordpress_api_key === currentAkismetKey;
+	const isStoredKey = wordpress_api_key === currentAkismetKey && !! wordpress_api_key;
```

Comment:
Since `wordpress_api_key` could be an empty string, we don't want to set `isStoredKey` as true in this case.